### PR TITLE
Introduce packages for controllers

### PR DESCRIPTION
**Pull request by xpivarc**
### What this PR does

This picks up where https://github.com/kubevirt/kubevirt/pull/12477 left. Moving each controller to its own package.

Fixes #

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/common/events.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package common
+
+const (
+	// FailedCreateVirtualMachineReason when a virtual machine is failed to be created.
+	//
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/common/events.go:5]: we can delete this, and the below one too

File: pkg/virt-controller/watch/common/events.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package common
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/common/events.go:1]: missing header

File: pkg/virt-controller/watch/common/events.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+const (
+	// FailedCreateVirtualMachineReason when a virtual machine is failed to be created.
+	//
+	FailedCreateVirtualMachineReason = "FailedCreate"
+	// SuccessfulCreateVirtualMachineReason when a virtual machine is successfully created.
+	//
+	SuccessfulCreateVirtualMachineReason = "SuccessfulCreate"
+	// FailedDeleteVirtualMachineReason when a virtual machine is failed to be deleted.
+	FailedDeleteVirtualMachineReason = "FailedDelete"
+	// SuccessfulDeleteVirtualMachineReason  when a virtual machine is successfully deleted.
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/common/events.go:12]:
```suggestion
	// SuccessfulDeleteVirtualMachineReason when a virtual machine is successfully deleted.
```

File: pkg/virt-controller/watch/replicaset/replicaset_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/replicaset/replicaset_suite_test.go:39]: This looks wrong :)

**Review by fossedihelm — COMMENTED** [review]
Some small comments below!
Thank you!

**Comment by xpivarc** [comment]
@fossedihelm PTAL

**Review by fossedihelm — COMMENTED** [review]
Thank you! The comment below is just for annotation :)
/lgtm

File: pkg/virt-controller/watch/pool/pool_test.go [hunk 5]
```diff
@@ -741,3 +742,7 @@ func DefaultPool(replicas int32) (*poolv1.VirtualMachinePool, *v1.VirtualMachine
 	virtcontroller.SetLatestApiVersionAnnotation(pool)
 	return pool, vm.DeepCopy()
 }
+
+func markVmAsReady(vm *v1.VirtualMachine) {
```
Comment (vladikr) [inline pkg/virt-controller/watch/pool/pool_test.go:746]: Is it a duplicate now?

**Comment by vladikr** [comment]
/approve
Looks good!

**Pull request merged**